packages/vite/src/node/plugins/html.ts

```diff
@@ -469,6 +470,19 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
           }
         }
 
+        await Promise.all(
+          styleUrls.map(async (styleUrl) => {
+            const resolvedUrl = await this.resolve(styleUrl, id)
+            if (resolvedUrl == null) {
+              config.logger.warnOnce(
+                `\n${styleUrl} doesn't exist at build time, it will remain unchanged to be resolved at runtime`
+              )
+            } else {
+              js += `\nimport ${JSON.stringify(styleUrl)}`
+            }
+          })
+        )
```

Comment:
This could change the order of the imports, no?
I think it is better to avoid the `styleUrls` array, and directly await in 342. Do you see a problem with doing it that way?

Reply:
walk function no suuport async/await 😭

Reply:
Oh, I see. Then we should at least avoid `Promise.all` here. If the `resolvedUrl` is null, aren't we still removing the tag in line 406? (`shouldRemove` is always true). Should `node.loc.start.offset, node.loc.end.offset` be also kept in the array to know if the tag should be removed?

Reply:
ok, But I don't understand why need to avoid `Promise.all` ?